test/scripts/base-host-check.sh

```diff
@@ -125,6 +125,151 @@ check_modularity() {
     done
 }
 
+check_hostname() {
+    echo "📗 Checking hostname"
+
+    # Verify that the hostname is set by running `hostname`
+    expected_hostname=$(jq -r '.blueprint.customizations.hostname' "${config}")
+    actual_hostname=$(hostname)
+
+    # we only emit a warning here since the hostname gets reset by cloud-init and we're not
+    # entirely sure how to deal with it yet on the service level
+    if [[ $actual_hostname != "${expected_hostname}" ]]; then 
+        echo "🟡 Hostname was not set: hostname=${actual_hostname} expected=${expected_hostname}"
+    else
+        echo "Hostname was set"
+    fi
```

Comment:
Could the check warn only if `cloud-init` service is enabled and fail in case there is no `cloud-init`?

Reply:
Only if the service is enabled, if the package is installed, or if it is configured in a certain way? There were a lot of ways and since we don't *currently* test in any ways where `cloud-init` isn't being used I felt it wasn't necessary.

Reply:
My idea was to warn only if the cloud-init service is enabled, which implies that it is installed. It could be configured not to set the hostname, but let's not complicate it with this option.

You are, of course, right that the effect in our current testing matrix is none. I was thinking more in the context of us not having to remember to fix this test case when we ever end up boot-testing image types without `cloud-init`.

FWIW, I'm OK with keeping the test as is for now.